pkg/apis/vpc/v1alpha1/subnet_types.go

```diff
@@ -104,6 +108,23 @@ type SubnetDHCPConfig struct {
 	Mode DHCPConfigMode `json:"mode,omitempty"`
 }
 
+// AdvancedConfig is Subnet advanced configuration.
+type AdvancedConfig struct {
+	// StaticIPAllocation configuration for subnet ports with VIF attachment.
+	// If this field is not set, the static IP pool allocation will be automatically
+	// enabled when the DHCP mode is DHCPDeactivated or not set and disabled when the DHCP
+	// mode is DHCPServer or DHCPRelay.
+	// The static IP pool allocation and DHCP mode cannot both be enabled simultaneously
+	// in subnet.
```

Comment:
This defaulting means that the value of `advanced_config.subnet_dhcp_config.enabled` could be true/false when the value in the CR is unset. When CCI API shows the field, it doesn't know whether the field should be explicitly set or not. For example, `enabled=false` in NSX API could be got from `enabled=nil` and `enabled=false` when DHCP is activivated; `enabled=true` in NSX API could be got from `enabled=nil` and `enabled=true` when DHCP is deactivated. Users may see different configurations in CCI and Supervisor API.

@jianjuns, does it violate the consistency you defined?

And to keep consistent defaulting behavior, in CCI we also need to make this field optional and determine its value based on DHCP configuration, instead of following NSX's default value.

Reply:
@tnqn : could you provide your suggested definition?